src/common/SteelSheets.hpp

```diff
@@ -0,0 +1,122 @@
+    ///
+    /// Printer use print sheet profile on the index 0 as a default so the method
+    /// return always at least 1 calibrated profile.
+    /// @return Return the count of the calibrated print sheet profiles.
+    static uint32_t NumOfCalibrated();
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine the name of the current active print sheet profile.
+    ///
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t ActiveSheetName(char *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Determine the name of the given print sheet profile.
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[out] buffer Buffer to store the print sheet profile
+    /// @param[in] length Size of the given buffer.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t SheetName(uint32_t index, char *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Rename the given print sheet profile.
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @param[in] buffer New name of the print sheet profile
+    /// @param[in] length Size of the given name.
+    /// @return Number of characters written to the buffer. Number will be
+    ///        always less than MAX_SHEET_NAME_LENGTH
+    static uint32_t RenameSheet(uint32_t index, char const *, uint32_t len);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Sets offset to currently selected sheet.
+    ///
+    /// @param[in] offset of the sheet
+    /// @return True when offset is set, False otherwise.
+    static bool SetZOffset(float offset);
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Gets offset of currently selected sheet.
+    ///
+    /// @return offset of the sheet
+    static float GetZOffset();
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Gets offset if sheet on index.
+    ///
+    /// @param[in] index Index of the sheet profile
+    /// @return offset of the sheet
+    static float GetSheetOffset(uint32_t index);
+
+private:
+    static Sheet getSheet(uint32_t index);
+    static bool setSheet(uint32_t index, Sheet sheet);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    /// @brief Updates probe offset variable in marlin
+    ///
+    /// @param[in] offset of the sheet
+    static void updateMarlin(float offset);
```

Comment:
I would propose to replace "offset" with "Z offset" everywhere in documentation.